.gitignore

```diff
@@ -8,4 +8,17 @@ core
 *~
 *.swp
 build
+CMakeCache.txt
```

Comment:
What is the preferred method? In source build or separate build dir?

Reply:
Personally, I prefer to have a different build directory. Typically just ${CMAKE_SOURCE_DIR}/build for me

Reply:
Ok. I only asked because if we strongly wanted to use a different build directory then we shouldn't add build generated files in the ignore list in order to not encourage that practice.